DRAFT.md

```diff
@@ -228,13 +228,52 @@ TODO
 
 TODO
 
-### Extra constructors and `emplace`
+### Supporting `operator()` `operator[]`
 
-TODO
+Neither `indirect` nor `polymorphic` support `operator()` or `operator[]`. These
+operators can't be implemented with `=default` and cannot, in general, be
+meaningfully implemented for a composite by aggregating over its components.
```

Comment:
I don't think this is a good explanation. an operator() can be made available if the (single) element of indirect has it, but the reason for not providing it is that it's always possible to just get the value and apply the call operator. That rationale doesn't apply to, say, operator<=>, because valueless indirects presumably order among the others? So you can't just <=> the value, you have to <=> the indirect. But for something like operator(), you can just always "dereference" the indirect.

Reply:
Oh, you can't spaceship valueless indirects either. Oh well.

Reply:
Is there a neater argument? Keen to merge unless what I've said is flat-out wrong.

Reply:
Well, it's plain incorrect to say that such an operator couldn't be done in terms of the component, because there's only one.

Reply:
I mean :

```c++
class Composite {
  indirect<A> a_;
  indirect<B> b_;
  
  auto operator() {
     // ???
   }
 };
 ```
 
 How can I reword my text to reflect this?

Reply:
Ahh.. yes. Something like this:
"There's no need for indirect/polymorphic to provide a function call or an indexing operator. Users who wish to do that can just access the value and call its operator. Furthermore, unlike comparisons, function calls or indexing operators don't compose further; for example <that example of yours>, a Composite wouldn't be able to automatically generate a composited operator() or an operator[]."

Reply:
Very nice. Applied.